src/collective/upload/static/main.js

```diff
@@ -186,32 +83,42 @@ $(document).ready(function() {
             });
 
             $(document).bind('drop', function (e) {
-                var url = $(e.originalEvent.dataTransfer.getData('text/html')).filter('img').attr('src');
-                if (url) {
-                    $.getImageData({
-                        url: url,
-                        server:'http://localhost:8080/Plone/@@jsonimageserializer?callback=?',
-                        success: function (img) {
-                            var canvas = document.createElement('canvas');
-                            canvas.width = img.width;
-                            canvas.height = img.height;
-                            if (canvas.getContext && canvas.toBlob) {
-                                canvas.getContext('2d').drawImage(img, 0, 0, img.width, img.height);
-                                canvas.toBlob(function (blob) {
-                                    $('#fileupload').fileupload('add', {files: [blob]});
-                                }, "image/jpeg");
-                            }
-                        },
-                        error: function(xhr, text_status){
-                            // Handle your error here
-                        }
-                    });
+              e.preventDefault();
+              var upload_image = function(img) {
+                var canvas = document.createElement('canvas');
+                canvas.width = img.width;
+                canvas.height = img.height;
+                canvas.getContext('2d').drawImage(img, 0, 0, img.width, img.height);
+                canvas.toBlob(function (blob) {
+                  $('#fileupload').fileupload('add', {files: [blob]});
+                }, "image/jpeg");
```

Comment:
this is just do fill the canvas preview, the original image is not hard-coded type